src/test/java/com/salesforce/dynamodbv2/mt/repo/MtDynamoDbTableDescriptionRepoTest.java

```diff
@@ -112,4 +118,49 @@ void testMtDynamoDbTableDescriptionPayPerRequestIsSet() throws InterruptedExcept
         TableUtils.waitUntilActive(localDynamoDb, fullTableName);
         DynamoDbTestUtils.assertPayPerRequestIsSet(fullTableName, localDynamoDb);
     }
+
+    @Test
+    void testListVirtualTableMetadatas() {
+        MtDynamoDbTableDescriptionRepo repo = mtDynamoDbTableDescriptionRepoBuilder.build();
+        String tenant1 = "1";
+        String tenant2 = "2";
+        String table1Gsi = "index";
+        String tableName1 = "table";
+        String tableName2 = "table2";
+        CreateTableRequest createReq1 = new CreateTableRequest()
+            .withTableName(tableName1)
+            .withKeySchema(new KeySchemaElement("id", KeyType.HASH))
+            .withGlobalSecondaryIndexes(new GlobalSecondaryIndex()
+                .withIndexName(table1Gsi)
+                .withKeySchema(new KeySchemaElement("secondary-id", KeyType.HASH)));
+        CreateTableRequest createReq2 = new CreateTableRequest()
+            .withTableName(tableName2)
+            .withKeySchema(new KeySchemaElement("id", KeyType.HASH));
+        MT_CONTEXT.withContext(tenant1, () ->
+            repo.createTable(createReq1));
+        MT_CONTEXT.withContext(tenant2, () ->
+            repo.createTable(createReq2));
+
+        ListMetadataResult returnedMetadatas = ((MtTableDescriptionRepo)repo).listVirtualTableMetadatas();
+
+        TenantTableMetadata tenantTable1 = new TenantTableMetadata(new TenantTable(tableName1, tenant1), createReq1);
+        TenantTableMetadata tenantTable2 = new TenantTableMetadata(new TenantTable(tableName2, tenant2), createReq2);
+        ListMetadataResult expected = new ListMetadataResult(
+            ImmutableList.of(tenantTable1, tenantTable2), null);
+
+        assertEquals(expected, returnedMetadatas);
+
+        returnedMetadatas = repo.listVirtualTableMetadatas(1, null);
+        expected = new ListMetadataResult(ImmutableList.of(tenantTable1), tenantTable1);
+        assertEquals(expected, returnedMetadatas);
```

Comment:
In the interest of making test code easier to understand and having having one assert per test method, I'd pull these into separate test methods.

You may also want to add some test cases for boundary conditions, e.g., empty repo, specifying the max value for limit, etc.

Reply:
makes sense, will submit a followup PR

Reply:
included in https://github.com/salesforce/mt-dynamo/pull/476